pump/storage/storage.go

```diff
@@ -674,28 +674,13 @@ func (a *Append) doGCTS(ts int64) {
 	}()
 
 	for {
-		nStr, err := a.metadata.GetProperty("leveldb.num-files-at-level0")
+		compacted, err := waitL0FilesCompacted(a.metadata, l0Trigger, 5*time.Second, &iter)
```

Comment:
In my opinion, having to pass the `&iter` reference just for cleaning up means we are extracting the wrong function. Ideally, the extracted function should know little about the original function so that it could later be reused in other cases (for example, where there's no Iterator involved).